type_lens/type_view.py

```diff
@@ -59,7 +59,9 @@ def __eq__(self, other: object) -> bool:
             return False
 
         if self.origin:
-            return bool(self.origin == other.origin and self.inner_types == other.inner_types)
+            self_origin = Union if self.is_union else self.origin
```

Comment:
Curious why this is needed?

Reply:
iirc due to Optional vs Union normalization across python versions. i could add a test for this to double check

Reply:
ah jk. we already have a test that fails without this on ParameterView for this reason

Reply:
Feels like ideally, we should normalise this at some point prior. Wdyt?